### PR TITLE
Add promise / await implementation

### PR DESCRIPTION
**Pull request by joelwurtz**
Replace #73

### CODE REVIEW

File: code/components/citizen-scripting-lua/src/LuaScriptRuntime.cpp [hunk 1]
```diff
@@ -194,7 +194,7 @@ class LuaScriptRuntime : public OMClass<LuaScriptRuntime, IScriptRuntime, IScrip
 
 LuaScriptRuntime::~LuaScriptRuntime()
 {
-
```
Comment (blattersturm) [inline code/components/citizen-scripting-lua/src/LuaScriptRuntime.cpp]: Seriously? Fix your editor. :/

**Comment by blattersturm** [comment]
In general, looks decent. Got any example usage?

**Comment by joelwurtz** [comment]
In mysql it would do the following, async will return promise instead of using callback :

```lua
local promise = MySQL.Async.fetchScalar("SELECT 1)
```

Then people would be able to use the next function

```lua
promise:next(function (result)
     -- Play with data
end)
```

Or there would be able to use Citizen.Await to wait (but in a unblocking way) for the result

```lua
local result = Citizen.Await(promise)
```

On the implementation side, i need to add an error callback / value to my exports, and it would work with something like this

```lua
function Async.fetchScalar(query, params)
    p = promise.new()

    export['mysql-async']:mysql_async_fetch_scalar(query, params, function(result, error)
        if error then
            p:reject(error)
        else
            p:resolve(result)
        end
    end)

    return p
end
```

File: code/components/citizen-scripting-lua/src/LuaScriptRuntime.cpp (hunk 1 above)
Comment (joelwurtz) [reply code/components/citizen-scripting-lua/src/LuaScriptRuntime.cpp]: Don't need fixing, it just try to be consistent, anyway readd them

**Comment by joelwurtz** [comment]
Made some tests with https://github.com/brouznouf/fivem-mysql-async/tree/feature/promise-await

works perfectly

**Pull request merged**